### PR TITLE
update syntax to be compatible with rubocop

### PR DESCRIPTION
**Pull request by b4ldr**
Hi 

i had a crack at addressing #72 its not very clean however it does most tests.  the only one i couldn't see how to fix easily was "Avoid comma after the last parameter of a method call" also whitespace however that is trivial to fix in an IDE so i haven't tried.  criticisms and comments welcome

Thanks John

### CODE REVIEW

**Comment by logicminds** [comment]
I still need to review, but I just ran into a bug with rspec-puppet https://github.com/rodjek/rspec-puppet/issues/530 and I wanted to make sure that retrospec takes this into account.

**Comment by b4ldr** [comment]
ok no rush, in the mean time i updated a few of the spec tests to conform with my changes

File: lib/retrospec/plugins/v1/plugin/generators/serializers/rspec_dumper.rb [hunk 1]
```diff
@@ -26,6 +26,19 @@ def logger
         @logger
       end
 
+      def dump_transform(d)
+        # This is a bit messy, ideally need to send a patch upstream to puppet
+        # https://github.com/puppetlabs/puppet/blob/master/lib/puppet/pops/visitor.rb#L34
```
Comment (logicminds) [inline lib/retrospec/plugins/v1/plugin/generators/serializers/rspec_dumper.rb]: What is this about exactly?  What would this patch look like?

**Comment by logicminds** [comment]
Can you document what the dump_transform is doing using [yard](http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md)  syntax.  I am going to start documenting all the methods so I will need  you to do the same.

Also need to rebase against master.

File: lib/retrospec/plugins/v1/plugin/generators/serializers/rspec_dumper.rb (hunk 1 above)
Comment (b4ldr) [reply lib/retrospec/plugins/v1/plugin/generators/serializers/rspec_dumper.rb]: the current do_dump object returns hash's represented in the ruby < 1.9 syntax.  this functions wraps that and munges the hash's to look like >= ruby 1.9 syntax

**Comment by b4ldr** [comment]
I have added some yard docs which describe the function to the best of my knowledge however i have to admit that i still don't completely understand the function this wraps which is why i have not sent a pull upstream

**Comment by logicminds** [comment]
I have been using this today and I like what I see.  However there are a few things I think we should fix as they might result in additional rubocop warnings.

1. closing `)` is not aligned

```
it do
    is_expected.to contain_ini_setting('enable_reports').with(
      ensure: 'present',
      section: 'user',
      setting: 'report',
      value: true,
      path: '$settings::confdir/puppet.conf',
      )
end  

```

2. Extra newline (:break) in params block.

```
let(:params) do
    {
      webhook: nil,
      channel: nil,
      # section: "user",

    }
end
```

**Comment by b4ldr** [comment]
the issues with params is in the template see here https://github.com/nwops/retrospec-templates/pull/14.  ill take a look at the other issues now

**Comment by b4ldr** [comment]
@logicminds closing bracket should be aligned now

**Pull request merged**